Assets/MixedRealityToolkit/Providers/BaseSpatialObserver.cs

```diff
@@ -55,6 +55,54 @@ public abstract class BaseSpatialObserver : BaseDataProvider<IMixedRealitySpatia
         /// </summary>
         protected IMixedRealitySpatialAwarenessSystem SpatialAwarenessSystem { get; private set; }
 
+        /// <summary>
+        /// Creates the spatial observer and handles the desired startup behavior.
+        /// </summary>
+        protected virtual void CreateObserver() { }
+
+        /// <summary>
+        /// Ensures that the spatial observer has been stopped and destroyed.
+        /// </summary>
+        protected virtual void CleanupObserver() { }
+
+        #region BaseService Implementation
+
+        /// <inheritdoc />
+        protected override void Dispose(bool disposing)
+        {
+            if (disposed) { return; }
```

Comment:
technically supposed to be 
```
if (disposed)
{
return;
}
```
right?

Reply:
Ah yeah, I can update this. Copied and pasted as it was

Reply:
Updated